File: eden/code-samples/eden.cs
/*
* Eden is full of beautiful islands...
*/

public class Island
{
  public string Name { get; private set; }
  public int Length { get; private set; }
  public int Width { get; private set; }
  public string[] Features { get; private set; }

  public Island(string name, int length, int width, string[] features)
  {
    Name = name;
    Length = length;
    Width = width;
    Features = features;
  }

  public int Area()
  {
    return calcArea();
  }

  private int calcArea()
  {
    return Length * Width;
  }
}
class Program
{
  static void Main(string[] args)
  {
    // ...one of them is named Elysian

    var elysian = new Island(
      "Elysian",
      300,
      350,
      new string[] { "Sunsets", "Beaches" }
    );
  }
}

File: eden/code-samples/eden.ts
/*
 * Eden is full of beautiful islands...
 */

interface IslandProps {
  name: string;
  length: number;
  width: number;
  features: Features;
}

type Features = string[];

export class Island {
  name: string;
  length: number;
  width: number;
  features: string[];

  constructor(props: IslandProps) {
    this.name = props.name;
    this.length = props.length;
    this.width = props.width;
    this.features = props.features;
  }

  get area(): number {
    return this.calcArea();
  }

  calcArea(): number {
    return this.length * this.width;
  }
}

// ...one of them is named Elysian

const elysian = new Island({
  name: 'Elysian',
  length: 300,
  width: 350,
  features: ['Sunsets', 'Beaches'],
});

let elysianArea = `Elysian Area: ${elysian.area}`;
